module/applications/actor/config/traits-config.mjs

```diff
@@ -0,0 +1,145 @@
+import SelectChoices from "../../../documents/actor/select-choices.mjs";
+import * as Trait from "../../../documents/actor/trait.mjs";
+import { filteredKeys } from "../../../utils.mjs";
+import BaseConfigSheet from "../api/base-config-sheet.mjs";
+
+/**
+ * Base application for selecting an actor's proficiencies.
+ */
+export default class TraitsConfig extends BaseConfigSheet {
+  /** @override */
+  static DEFAULT_OPTIONS = {
+    classes: ["trait-selector"],
+    trait: null,
+    position: {
+      width: 600
+    }
+  };
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  static PARTS = {
+    traits: {
+      template: "systems/dnd5e/templates/actors/config/traits-config.hbs"
+    }
+  };
+
+  /* -------------------------------------------- */
+  /*  Properties                                  */
+  /* -------------------------------------------- */
+
+  /** @override */
+  get title() {
+    return Trait.traitLabel(this.options.trait);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Label used for the "other" category.
+   * @type {string}
+   */
+  get otherLabel() {
+    return game.i18n.localize("DND5E.ProficiencyOther");
+  }
+
+  /* -------------------------------------------- */
+  /*  Rendering                                   */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  _initializeApplicationOptions(options) {
+    options = super._initializeApplicationOptions(options);
+    if ( !CONFIG.DND5E.traits[options.trait] ) throw new Error(
+      `Cannot instantiate TraitsConfig with a trait not defined in CONFIG.DND5E.traits: ${options.trait}.`
+    );
+    options.uniqueId = `${options.trait}-${options.document.uuid}`;
+    return options;
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _preparePartContext(partId, context, options) {
+    context = await super._preparePartContext(partId, context, options);
+    context.keyPath = Trait.actorKeyPath(this.options.trait);
+    context.data = foundry.utils.getProperty(this.document.toObject(), context.keyPath);
```

Comment:
Missed this before. Should be safe to do, we are not expecting `_processChoices` to mutate `data`, right?

```suggestion
    context.data = foundry.utils.getProperty(this.document._source, context.keyPath);
```